src/filehandle.ts

```diff
@@ -15,6 +15,12 @@ export interface FilehandleOptions {
 }
 
 export interface GenericFilehandle {
-  read(buf: Buffer, offset: number, length: number, position: number, opts?: FilehandleOptions): Promise<number>
+  read(
+    buf: Buffer,
+    offset: number,
+    length: number,
+    position: number,
+    opts?: FilehandleOptions,
+  ): Promise<{ bytesRead: number; buffer: Buffer }>
   readFile(options?: FilehandleOptions): Promise<Buffer | string>
```

Comment:
```suggestion
   readFile(options?: FilehandleOptions | string): Promise<Buffer | string>
```
Not related to these changes, but I just realized this can be string. As with the Node readFile, if it's a string it assumes it's the encoding.

Reply:
indeed, looks like the modules already have that in the interface but the abstract interface needed it.